src/binder/binder_util.cpp

```diff
@@ -90,13 +90,20 @@ void BinderUtil::CheckAndTryPromoteType(const common::ManagedPointer<parser::Con
             break;
           }
           case type::TypeId::TINYINT: {
+            size_t size;
```

Comment:
We used to double-template (template calls a template) this code, but I think we went back to this style for readability.

We could conceivably macro the whole thing instead. It is a little repetitive. No strong opinions though.